components/locid/src/extensions/other/mod.rs

```diff
@@ -0,0 +1,127 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Other Use Extensions is a list of extensions other than unicode,
+//! transform or private.
+//!
+//! Those extensions are treated as a pass-through, and no Unicode related
+//! behavior depends on them.
+//!
+//! The main struct for this extension is [`Other`] which is a list of [`Keys`].
+//!
+//! # Examples
+//!
+//! ```
+//! use icu::locid::Locale;
+//! use icu::locid::extensions::other::Other;
+//!
+//! let mut loc: Locale = "en-US-a-foo-faa".parse()
+//!     .expect("Parsing failed.");
+//! ```
+//!
+//! [`Keys`]: Key
+mod key;
+use alloc::boxed::Box;
+
+use crate::parser::ParserError;
+use alloc::vec::Vec;
+use core::iter::Peekable;
+
+pub use key::Key;
+
+/// A list of [`Other Use Extensions`] as defined in [`Unicode Locale
+/// Identifier`] specification.
+///
+/// Those extensions are treated as a pass-through, and no Unicode related
+/// behavior depends on them.
+///
+/// # Examples
+///
+/// ```
+/// use icu::locid::extensions::other::{Other, Key};
+///
+/// let key1: Key = "foo".parse()
+///     .expect("Failed to parse a Key.");
+/// let key2: Key = "bar".parse()
+///     .expect("Failed to parse a Key.");
+///
+/// let other = Other::from_vec_unchecked('a' as u8, vec![key1, key2]);
+/// assert_eq!(&other.to_string(), "-a-foo-bar");
+/// ```
+///
+/// [`Other Use Extensions`]: https://unicode.org/reports/tr35/#other_extensions
+/// [`Unicode Locale Identifier`]: https://unicode.org/reports/tr35/#Unicode_locale_identifier
+#[derive(Clone, PartialEq, Eq, Debug, Default, Hash, PartialOrd, Ord)]
+pub struct Other((u8, Box<[Key]>));
+
+impl Other {
+    /// A constructor which takes a pre-sorted list of [`Key`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locid::extensions::other::{Other, Key};
+    ///
+    /// let key1: Key = "foo".parse()
+    ///     .expect("Failed to parse a Key.");
+    /// let key2: Key = "bar".parse()
+    ///     .expect("Failed to parse a Key.");
+    ///
+    /// let other = Other::from_vec_unchecked('a' as u8, vec![key1, key2]);
+    /// assert_eq!(&other.to_string(), "-a-foo-bar");
+    /// ```
+    pub fn from_vec_unchecked(ext: u8, input: Vec<Key>) -> Self {
+        Self((ext, input.into_boxed_slice()))
+    }
+
+    pub(crate) fn try_from_iter<'a>(
+        ext: u8,
```

Comment:
Question: Is there a point at which you verify that `ext` is ascii-alphabetic?

If yes:
- Nit (optional): Add a `debug_assert!` in this function

If no:
- Issue: Add such a check and return an error code.

Reply:
Thanks for pointing this out, I've added a check for this in `extensions/mod.rs`.